cypress/e2e/tests/migration/migration-waves/filter.test.ts

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright © 2021 the Konveyor Contributors (https://konveyor.io/)
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/// <reference types="cypress" />
+
+import {
+    login,
+    clickByText,
+    applySearchFilter,
+    preservecookies,
+    deleteByList,
+    createMultipleMigrationWaves,
+} from "../../../../utils/utils";
+import { button, name, clearAllFilters, SEC } from "../../../types/constants";
+
+import * as data from "../../../../utils/data_utils";
+import { MigrationWave } from "../../../models/migration/migration-waves/migration-wave";
+
+let migrationWavesList: Array<MigrationWave> = [];
+let invalidSearchInput = String(data.getRandomNumber());
+
+describe(["@tier2"], "Migration waves filter validations", function () {
+    before("Login and Create Test Data", function () {
+        // Peform login
+        login();
+
+        // Create multiple migration waves
+        migrationWavesList = createMultipleMigrationWaves(3);
+    });
+
+    after("Perform test data clean up", function () {
+        // Delete the business services
+        deleteByList(migrationWavesList);
+    });
+
+    beforeEach("Persist session", function () {
+        // Save the session and token cookie for maintaining one login session
+        preservecookies();
+    });
+
+    it("Name filter validations", function () {
+        MigrationWave.open();
+
+        // Enter an existing display name substring and assert
+        let validSearchInput = migrationWavesList[0].name.substring(0, 3);
+        applySearchFilter(name, validSearchInput);
+        cy.get("td", { timeout: 12 * SEC }).should("contain", migrationWavesList[0].name);
+
+        if (migrationWavesList[1].name.indexOf(validSearchInput) >= 0) {
+            cy.get("td", { timeout: 12 * SEC }).should("contain", migrationWavesList[1].name);
+        }
+        clickByText(button, clearAllFilters);
+
+        // Enter an existing exact name and assert
+        applySearchFilter(name, migrationWavesList[1].name);
+        cy.get("td", { timeout: 12 * SEC }).should("contain", migrationWavesList[1].name);
+        cy.get("td", { timeout: 12 * SEC }).should("not.contain", migrationWavesList[0].name);
+        clickByText(button, clearAllFilters);
+
+        // Enter a non-existing name substring and apply it as search filter
+        applySearchFilter(name, invalidSearchInput);
```

Comment:
The `invalidSearchInput` variable is only used here so it doesn't make really sense to store a value in a variable that is used only once.
You can just put the value.

`applySearchFilter(name, String(data.getRandomNumber()));`